lib/realtime/tenants/connect.ex

```diff
@@ -117,18 +130,22 @@ defmodule Realtime.Tenants.Connect do
   def init(%{tenant_id: tenant_id} = state) do
     Logger.metadata(external_id: tenant_id, project: tenant_id)
 
-    with %Tenant{} = tenant <- Tenants.get_tenant_by_external_id(tenant_id),
+    with %Tenant{} = tenant when not is_nil(tenant) <-
```

Comment:
```suggestion
    with %Tenant{} = tenant <-
```

`tenant` cannot be nil here, since it has to match a struct.